### PR TITLE
issue-243: Fixes admin bug to reflect changes automatically

### LINKED ISSUE #243: "Admin Mode" doesn't work properly
The toggling the checkbox doesn't reflect to UI immediately.

### PR DESCRIPTION
**Pull request by harunurhan**
* Fixes automatic change detection on disabled fields
  * Fixes enabling hidden fields on admin mode

(Closes #243)
Signed-off-by: Zacharias Zacharodimos <zacharias.zacharodimos@cern.ch>

Signed-off-by: harunurhan <harunurhan17@gmail.com>

### CODE REVIEW

File: src/primitive-field/primitive-field.component.ts [hunk 1]
```diff
@@ -105,7 +109,7 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent {
   }
 
   get disabled(): boolean {
-    return this.schema.disabled && !this.appGlobalsService.adminMode;
+    return this.schema.disabled && !this.appGlobalsService.adminMode ;
```
Comment (jmartinm) [inline src/primitive-field/primitive-field.component.ts]: extra space

File: src/sub-record/sub-record.component.ts [hunk 2]
```diff
@@ -57,6 +57,7 @@ export class SubRecordComponent extends AbstractTrackerComponent implements OnCh
   constructor(public jsonStoreService: JsonStoreService,
     public appGlobalsService: AppGlobalsService,
     public tabsUtilService: TabsUtilService,
+    public changeDetectorRef: ChangeDetectorRef,
```
Comment (jmartinm) [inline src/sub-record/sub-record.component.ts]: unused

File: src/object-field/object-field.component.ts [hunk 3]
```diff
@@ -61,4 +61,7 @@ export class ObjectFieldComponent extends AbstractFieldComponent {
     this.keysStoreService.deleteKey(this.pathString, name);
   }
 
+  isDisabled(key): boolean {
+    return this.schema.properties[key].disabled && !this.appGlobalsService.adminMode ;
```
Comment (jmartinm) [inline src/object-field/object-field.component.ts]: Extra space

**Comment by harunurhan** [comment]
@jmartinm I am aware of the `tree-menu` behaviour, I didn't think it was important so didn't do it to keep code cleaner. but If it's not OK, I can also implement it in this PR.

**Comment by jmartinm** [comment]
Create follow up issue https://github.com/inveniosoftware-contrib/ng2-json-editor/issues/309

**Pull request merged**